examples/flutter_gallery/lib/demo/rally/charts/pie_chart.dart

```diff
@@ -0,0 +1,249 @@
+        child: Center(
+          child: Column(
+            mainAxisAlignment: MainAxisAlignment.center,
+            children: <Widget>[
+              Text(
+                centerLabel,
+                style: labelTextStyle,
+              ),
+              Text(
+                usdWithSignFormat.format(centerAmount),
+                style: Theme.of(context).textTheme.headline,
+              ),
+            ],
+          ),
+        ),
+      ),
+    );
+  }
+}
+
+class _RallyPieChartOutlineDecoration extends Decoration {
+  const _RallyPieChartOutlineDecoration(
+      {this.maxFraction, this.total, this.segments});
+
+  final double maxFraction;
+  final double total;
+  final List<RallyPieChartSegment> segments;
+
+  @override
+  BoxPainter createBoxPainter([VoidCallback onChanged]) {
+    return _RallyPieChartOutlineBoxPainter(
+      maxFraction: maxFraction,
+      wholeAmount: total,
+      segments: segments,
+    );
+  }
+}
+
+class _RallyPieChartOutlineBoxPainter extends BoxPainter {
+  _RallyPieChartOutlineBoxPainter(
+      {this.maxFraction, this.wholeAmount, this.segments});
+
+  final double maxFraction;
+  final double wholeAmount;
+  final List<RallyPieChartSegment> segments;
+
+  @override
+  void paint(Canvas canvas, Offset offset, ImageConfiguration configuration) {
+    // Create two padded reacts to draw arcs in: one for colored arcs and one for
+    // inner bg arc.
+    const double strokeWidth = 4.0;
+    final double outerRadius =
+        min(configuration.size.width, configuration.size.height) / 2;
+    final Rect outerRect = Rect.fromCircle(
+        center: configuration.size.center(Offset.zero),
+        radius: outerRadius - strokeWidth * 3.0);
+    final Rect innerRect = Rect.fromCircle(
+        center: configuration.size.center(Offset.zero),
+        radius: outerRadius - strokeWidth * 4.0);
+
+    // Paint each arc with spacing.
+    double cumulativeSpace = 0.0;
+    double cumulativeTotal = 0.0;
+    const double wholeRadians = 2.0 * pi;
+    const double spaceRadians = wholeRadians / 180.0;
+    final double wholeMinusSpacesRadians =
+        wholeRadians - (segments.length * spaceRadians);
+    for (RallyPieChartSegment segment in segments) {
+      final Paint paint = Paint()..color = segment.color;
+      final double start = maxFraction *
+              ((cumulativeTotal / wholeAmount * wholeMinusSpacesRadians) +
+                  cumulativeSpace) -
+          pi / 2.0;
+      final double sweep =
+          maxFraction * (segment.value / wholeAmount * wholeMinusSpacesRadians);
+      canvas.drawArc(outerRect, start, sweep, true, paint);
+      cumulativeTotal += segment.value;
+      cumulativeSpace += spaceRadians;
+    }
+
+    // Paint any remaining space black (e.g. budget amount remaining).
+    final double remaining = wholeAmount - cumulativeTotal;
+    if (remaining > 0) {
+      final Paint paint = Paint()..color = Colors.black;
+      final double start = maxFraction *
```

Comment:
this looks similar to the code above, can it be refactored and reused?

Reply:
Refactored it out to 3 small methods. Made it easier to understand I think!